core/lib/spree/testing_support/factories/payment_factory.rb

```diff
@@ -6,7 +6,7 @@
 FactoryGirl.define do
   factory :payment, aliases: [:credit_card_payment], class: Spree::Payment do
     association(:payment_method, factory: :credit_card_payment_method)
-    association(:source, factory: :credit_card)
+    source { create(:credit_card, user: order.user) }
```

Comment:
I don't believe that will work, as `order` will be evaluated at the time of factory definition, as opposed to when a factory instance is created.

    [...]/payment_factory.rb:9:in `block (2 levels) in <top (required)>': undefined method `user' for #<FactoryGirl::Declaration::Implicit:0x007f0bfda556d0> (NoMethodError)